ctapipe/tools/make_irf.py

```diff
@@ -0,0 +1,385 @@
+            self.log.debug(self.fov_offset_bins)
+            hdus.append(
+                create_aeff2d_hdu(
+                    effective_area[..., np.newaxis],  # +1 dimension for FOV offset
+                    self.true_energy_bins,
+                    self.fov_offset_bins,
+                    extname="EFFECTIVE AREA" + label,
+                )
+            )
+            edisp = energy_dispersion(
+                self.signal[mask],
+                true_energy_bins=self.true_energy_bins,
+                fov_offset_bins=self.fov_offset_bins,
+                migration_bins=self.energy_migration_bins,
+            )
+            hdus.append(
+                create_energy_dispersion_hdu(
+                    edisp,
+                    true_energy_bins=self.true_energy_bins,
+                    migration_bins=self.energy_migration_bins,
+                    fov_offset_bins=self.fov_offset_bins,
+                    extname="ENERGY_DISPERSION" + label,
+                )
+            )
+        # Here we use reconstructed energy instead of true energy for the sake of
+        # current pipelines comparisons
+        bias_resolution = energy_bias_resolution(
+            self.signal[self.signal["selected"]],
+            self.reco_energy_bins,
+            energy_type="reco",
+        )
+        hdus.append(fits.BinTableHDU(bias_resolution, name="ENERGY_BIAS_RESOLUTION"))
+
+        # Here we use reconstructed energy instead of true energy for the sake of
+        # current pipelines comparisons
+        ang_res = angular_resolution(
+            self.signal[self.signal["selected_gh"]],
+            self.reco_energy_bins,
+            energy_type="reco",
+        )
+        hdus.append(fits.BinTableHDU(ang_res, name="ANGULAR_RESOLUTION"))
+
+        background_rate = background_2d(
+            self.background[self.background["selected_gh"]],
+            self.reco_energy_bins,
+            fov_offset_bins=self.bkg_fov_offset_bins,
+            t_obs=self.obs_time * u.Unit(self.obs_time_unit),
+        )
+        hdus.append(
+            create_background_2d_hdu(
+                background_rate,
+                self.reco_energy_bins,
+                fov_offset_bins=self.bkg_fov_offset_bins,
+            )
+        )
+
+        psf = psf_table(
+            self.signal[self.signal["selected_gh"]],
+            self.true_energy_bins,
+            fov_offset_bins=self.fov_offset_bins,
+            source_offset_bins=self.source_offset_bins,
+        )
+        hdus.append(
+            create_psf_table_hdu(
+                psf,
+                self.true_energy_bins,
+                self.source_offset_bins,
+                self.fov_offset_bins,
+            )
+        )
+
+        hdus.append(
+            create_rad_max_hdu(
+                self.theta_cuts_opt["cut"].reshape(-1, 1),
+                self.true_energy_bins,
+                self.fov_offset_bins,
+            )
+        )
+
```

Comment:
For debugging and benchmarking, it might be nice to add a few more historgrams in the output files (all vs energy and offset):
* N_on and N_off counts (maybe even N_excess and Significance), which can help to see where  limiting factors of background and minimum counts are taken into account.
* background rate _per particle species_ (same as the background2d already there, but for electrons and protons separately)
* Effective area for protons and electrons separately